posts/inside-rust/2020-06-24-lto-improvements.md

```diff
@@ -0,0 +1,89 @@
+---
+layout: post
+title: "Disk space and LTO improvements"
+author: Eric Huss
+description: "Disk space and LTO improvements"
+team: the Cargo team <https://www.rust-lang.org/governance/teams/dev-tools#cargo>
+---
+
+Thanks to the work of [Nicholas Nethercote] and [Alex Crichton], there have been some recent improvements that reduce the size of compiled libraries, and improves the compile-time performance, particularly when using LTO. This post dives into some of the details of what changed, and an estimation of the benefits.
+
+These changes have been added incrementally over the past three months, with the latest changes landing just a few days ago on the nightly channel. The bulk of the improvements will be found in the 1.46 stable release. It would be great for any projects that use LTO to test it out on the nightly channel and report any issues that arise.
+
+[Nicholas Nethercote]: https://github.com/nnethercote
+[Alex Crichton]: https://github.com/alexcrichton/
+
+## Background
+
+When compiling a library, `rustc` saves the output in an `rlib` file which is an [archive file]. This has historically contained these two things (among others):
```

Comment:
Not two but three things: there is metadata as well.

Reply:
Maybe the metadata is covered by the "(among others)", but given that it's the only thing in "(among others)", it's probably just worth mentioning outright.

Reply:
Yea, it seemed awkward when I wrote it.  I was attempting to focus on just the things that changed. I went ahead and just added the metadata as the 3rd major part.